src/exercises/2-4-State_encapsulation/final/Dialog.md

```diff
@@ -0,0 +1,51 @@
+```jsx
+import {
+  useDialogState,
+  Dialog,
+  DialogContent,
+  DialogBackdrop,
+  DialogHeader,
+  DialogBody,
+  DialogFooter,
+  DialogCloseButton
+} from './final/Dialog';
+const [name, setName] = React.useState('');
+const [draftName, setDraftName] = React.useState('');
+const dialog = useDialogState();
+<>
+  <Dialog {...dialog}>
+    <DialogBackdrop {...dialog} />
+    <DialogContent>
+      <DialogCloseButton {...dialog} />
```

Comment:
Hmm, I thought we would put the button into the Header to make the example of impossible state: No header but close button should be there. Should we?

Reply:
We can't expect everyone to make this “mistake”, also this is how Chakra does it — the close button and the header are independent. So there's no right way to do it.

Reply:
But we don’t have to follow chakra here and we can do the mistake ourselves while explaining

Reply:
That's not what I meant. It's a valid API choice and showing it as a mistake isn't a good idea. It doesn't make the state impossible either.

Reply:
Okay let's merge